lib/utils/embeddings/db.ts

```diff
@@ -0,0 +1,50 @@
+import weaviate from 'weaviate-ts-client'
+import deleteRepo from './delete'
+import addRepo from './embed'
+import checkIndexExists from './exists'
+import search from './query'
+
+export type WeaviateClient = ReturnType<typeof weaviate.client>
+
+export type WeaviateConfig = {
+	client: WeaviateClient
+	userId: string
+	indexName: string
+}
+
+export default class Weaviate {
+	client: WeaviateClient
+	config: WeaviateConfig
+
+	constructor(userId: string, indexName: string = 'CodeSearch') {
+		this.client = weaviate.client({
+			scheme: process.env.WEAVIATE_SCHEME,
+			host: process.env.WEAVIATE_HOST
+		})
+
+		checkIndexExists(this.client, indexName)
+
+		this.config = {
+			client: this.client,
+			userId: userId,
+			indexName: indexName
+		}
+	}
+
+	async embedRepo(repoUrl: string, branch: string) {
```

Comment:
This class makes for super clean usage

@arihanv would you be up to adjusting `embedRepo` so it upserts (or fails) when the repo+customer are already in the Weaviate instance?